packages/graphql/src/schema/resolvers/field/id.ts

```diff
@@ -18,17 +18,34 @@
  */
 
 import { GraphQLResolveInfo } from "graphql";
-import defaultFieldResolver from "./defaultField";
-import { isNeoInt } from "../../utils/utils";
+import { defaultFieldResolver } from "./defaultField";
+import { isNeoInt } from "../../../utils/utils";
+import { Context } from "../../../types";
 
-function numerical(source, args, context, info: GraphQLResolveInfo) {
+export function idResolver(source, args, context: Context, info: GraphQLResolveInfo) {
     const value = defaultFieldResolver(source, args, context, info);
 
+    if (Array.isArray(value)) {
+        return value.map((v) => {
+            if (isNeoInt(v)) {
+                return v.toNumber();
+            }
+
+            if (typeof v === "number") {
+                return v.toString(10);
+            }
+
+            return v;
```

Comment:
Consider extracting this into a separate function to remove duplicate code, so `idResolver` looks something like:

```ts
const value = defaultFieldResolver(source, args, context, info);
if (Array.isArray(value)){
  return value.map((v)=> numberResolver(v));
}
return numberResolver(v)
```